op-node/eth/label.go

```diff
@@ -0,0 +1,19 @@
+package eth
+
+type BlockLabel string
+
+const (
```

Comment:
Perhaps this type in upstream is useful?
https://github.com/ethereum/go-ethereum/blob/6d711f0c001ccb536c5ead8bd5d07828819e7d61/rpc/types.go#L61

Reply:
I don't particularly like using the negative integers to present the special values, and if we ever end up adding our own it'll be less flexible. That said, maybe in a future PR we can consolidate the `ByLabel/ByNumber/ByHash/ByRpcNumber` variants of essentially the same thing all into one function which takes some interface that allows the argument to be encoded for the request, which can then be implemented by any of these types.

Reply:
Opened ENG-2673 to address this in some later PR.